src/utils/aadV2TokenProvider.ts

```diff
@@ -0,0 +1,278 @@
+import { Clipboard, commands, env, Uri, window } from 'vscode';
+import * as Constants from '../common/constants';
+import { HttpRequest } from '../models/httpRequest';
+import { HttpClient } from './httpClient';
+import { EnvironmentVariableProvider } from './httpVariableProviders/environmentVariableProvider';
+
+/*
+ AppId provisioned to allow users to explicitly consent to permissions that this app can call
+*/
+export const AadV2TokenProviderClientId = "07f0a107-95c1-41ad-8f13-912eab68b93f";
+
+export class AadV2TokenProvider {
+    private readonly _httpClient: HttpClient;
+    private readonly clipboard: Clipboard;
+
+    public constructor() {
+        this._httpClient = new HttpClient();
+        this.clipboard = env.clipboard;
+    }
+
+    public async AcquireToken(name : string): Promise<string> {
+
+        const authParams = new AuthParameters();
+        await authParams.ParseName(name);
+
+        if (!authParams.forceNewToken) {
+            const tokenEntry = AadV2TokenCache.GetToken(authParams.GetCacheKey());
+            if (tokenEntry && tokenEntry.SupportScopes(authParams.scopes)) {
```

Comment:
Simply use the optional chaining syntax `?.` and the code will be implied as following
```typescript
if (tokenEntry?.SupportScopes(authParams.scopes))
```